bond/commands/upgrade.py

```diff
@@ -41,13 +41,17 @@ def do_upgrade(bondid, version_obj):
         pass
     for _ in range(120):
         time.sleep(1)
-        rsp = bond.proto.get(bondid, topic="sys/upgrade")
-        progress = rsp["b"]["progress"]
-        assert rsp["s"] in (200, 204)
-        if rsp["s"] == 204 or progress == 1000:
-            print("Upgrade installed.")
-            break
-        print(f"Progress: {progress / 10}%")
+        try:
+            rsp = bond.proto.get(bondid, topic="sys/upgrade")
+            progress = rsp["b"]["progress"]
+            print(f"Progress: {progress / 10}%")
+            if rsp["s"] == 204 or progress == 1000:
+                print("Upgrade installed.")
+                break
+        except:
```

Comment:
This should be reduced to catch only the appropriate class of exceptions. Moreover, the try/catch should be wrapped more tightly around the code which is failing. Presumably it is bond.proto.get that is raising, not print or one of the conditionals on line 48.

The broad except (many lines, any exception) makes the inevitable typo in those interveneing lines harder to debug,.

Reply:
Okay thanks! Done